### ISSUE TITLE
Parse error on type asserts.

### ISSUE THREAD

**Issue by MineBill** [obiwan87/odin-intellij#33]
Consider this:
```odin
package unions
import "core:fmt"

Value :: union {
    string,
    int,
    f32,
}

main :: proc() {
    a: Value = 2

    as_int := &a.(int) // <--- HERE
    as_int^ = 3

    fmt.printfln("a: %v", a)
}
```

Taking the address while doing a type assert will produce the following error:
```
OdinTokenType.IDENTIFIER_TOKEN expected, got '('
```

**Comment by obiwan87** [comment]
Ah thanks. Missed that one. Should be failing with any unary operator (try +, -)?

**Comment by obiwan87** [comment]
It's now fixed with commit 3dc3cad
